packages/astro/src/core/errors/errors-data.ts

```diff
@@ -1196,6 +1196,18 @@ export const i18nNotEnabled = {
 	hint: 'See https://docs.astro.build/en/guides/internationalization for a guide on setting up i18n.',
 } satisfies ErrorData;
 
+/**
+ * @docs
+ * @description
+ * The path doesn't contain any locale. The user should use `pathHasLocale` first.
```

Comment:
```suggestion
 * An i18n utility requires a `locale` property, but the path doesn't contain one. Use `pathHasLocale` first to check for a `locale` before using i18n utilities.
```

Reply:
The suggestion is misleading. The function accepts a `path` and the path must contain a locale .e.g `/en/slug`. The user doesn't provide locale directly

Reply:
I think we're saying the same thing? I'm just trying to better understand. I'm not suggesting that a user provide a locale directly, but (if I understand this correctly) they should add some logic with `pathHasLocal` to perform a check for whether or not a path in included before running logic that takes `path` and assumes it has a locale?

Reply:
> An i18n utility requires a `locale` property

This is the part that is confusing to me. There's no property involved 😅 I will make an example, maybe it will help

```js
`getLocaleFromPath("/slug")` // error, because there's not locale in the path, this error is thrown
`getLocaleFromPath("/en/slug")` // correct, no error, the user gets the correct value
```

Reply:
Ah, I think I get why my version sounded confusing! I was jumping ahead to explain the later problem that someone would have, that an i18n utility wasn't going to work. (from the `message`)

I was trying to provide a bit more content about the situation in which this problem of missing locale would come up, but maybe that's too far a jump. This content shows up under the heading "What went wrong?" in docs, so I was trying to make this explain a little more.

Here's another attempt, and if I still fail, then we keep the original until someone complains it's too short and doesn't explain enough!  :sweat_smile: 

> An i18n utility tried to use the locale from a URL path that does not contain one. You can prevent this error by using `pathHasLocale` to check URLs for a `locale` first before using i18n utilities.